src/ert/_c_wrappers/enkf/res_config.py

```diff
@@ -265,6 +265,7 @@ def _alloc_from_dict(self, config_dict):
         )
         jobs = []
         for job_name, args in config_dict.get(ConfigKeys.FORWARD_MODEL, []):
+            job_name = self.substitution_list.substitute(job_name)
             try:
                 job = copy.deepcopy(self.installed_jobs[job_name])
             except KeyError as err:
```

Comment:
in line 455, github says:

> `Use 'self != other' instead of 'not self == other'`

Reply:
That is going to do an infinite recursion and so its just wrong. This is being fixed by #4743